webapp/src/components/calendar/fullCalendar.tsx

```diff
@@ -0,0 +1,206 @@
+        const {event} = eventProps
+        return (
+            <div>
+                <div className='octo-icontitle'>
+                    { event.extendedProps.icon ? <div className='octo-icon'>{event.extendedProps.icon}</div> : undefined }
+                    <div
+                        className='fc-event-title'
+                        key='__title'
+                    >{event.title || intl.formatMessage({id: 'KanbanCard.untitled', defaultMessage: 'Untitled'})}</div>
+                </div>
+                {visiblePropertyTemplates.map((template) => (
+                    <Tooltip
+                        key={template.id}
+                        title={template.name}
+                    >
+                        <PropertyValueElement
+                            board={board}
+                            readOnly={true}
+                            card={cards.find((o) => o.id === event.id) || cards[0]}
+                            contents={[]}
+                            comments={[]}
+                            propertyTemplate={template}
+                            showEmptyPlaceholder={false}
+                        />
+                    </Tooltip>
+                ))}
+            </div>
+        )
+    }
+
+    const eventClick = (eventProps: EventClickArg) => {
+        if (!readonly) {
+            const {event} = eventProps
+            props.showCard(event.id)
+        }
+    }
+
+    const eventChange = (eventProps: EventChangeArg) => {
+        const {event} = eventProps
+        if (!event.start) {
+            return
+        }
+        if (!event.end) {
+            return
+        }
+
+        const startDate = new Date(event.start.getTime())
+        const endDate = new Date(event.end.getTime())
+        const dateProperty = createDatePropertyFromCalendarDates(startDate, endDate)
+        const card = cards.find((o) => o.id === event.id)
+        if (card && dateDisplayProperty) {
+            mutator.changePropertyValue(card, dateDisplayProperty.id, JSON.stringify(dateProperty))
+        }
+    }
+
+    const onNewEvent = (args: {start: Date, end: Date}) => {
+        const dateProperty = createDatePropertyFromCalendarDates(args.start, args.end)
+
+        const properties: Record<string, string> = {}
+        if (dateDisplayProperty) {
+            properties[dateDisplayProperty.id] = JSON.stringify(dateProperty)
+        }
+
+        props.addCard(properties)
+    }
+
+    const toolbar = {
+        left: 'title',
+        center: '',
+        right: 'dayGridWeek dayGridMonth prev,today,next',
+    }
+
+    const buttonText = {
+        today: intl.formatMessage({id: 'calendar.today', defaultMessage: 'TODAY'}),
+        month: intl.formatMessage({id: 'calendar.month', defaultMessage: 'Month'}),
+        week: intl.formatMessage({id: 'calendar.week', defaultMessage: 'Week'}),
+    }
+
+    return (
+        <div
+            className='CalendarContainer'
+        >
+            <FullCalendar
```

Comment:
Here is a long list of not-memoized properties, it doesn't matter in the first level, because FullCalendar is not a PureComponent, but can affects in the re-rendering of other more internal components. I would prefer to memoize more things, like the `buttonText`, the `toolbar`, or the callbacks.

Reply:
I think I got all of these.